bentoml/_internal/server/__init__.py

```diff
@@ -107,12 +146,35 @@ def serve_development(
         watchers,
         sockets=list(circus_socket_map.values()),
     )
+    tracking_thread, stop_thread_event = scheduled_track(
+        current_pid,
+        event_properties=get_scheduled_event_properties(
+            production=False,
+            bento_identifier=bento_identifier,
+            serve_info=serve_info,
+        ),
+    )
     _ensure_prometheus_dir()
 
-    arbiter.start(
-        cb=lambda _: logger.info(  # type: ignore
-            f'Starting development BentoServer from "{bento_identifier}" '
-            f"running on http://{host}:{port} (Press CTRL+C to quit)"
+    try:
+        tracking_thread.start()
+        arbiter.start(
+            cb=lambda _: logger.info(  # type: ignore
+                f'Starting development BentoServer from "{bento_identifier}" '
+                f"running on http://{host}:{port} (Press CTRL+C to quit)"
+            ),
+        )
+    finally:
+        stop_thread_event.set()
+
+    atexit.register(
```

Comment:
move this to before the `arbiter.start` at l158

Reply:
done